### ISSUE TITLE
Chebfun2 enters infinite loop for functions that cannot be resolved

### ISSUE THREAD

**Issue by gradywright** [chebfun/chebfun#1066]
The following causes the chebfun2 `constructor` to enter an infinite loop.
```
f = chebfun2(@(x,y) sign(x).*sign(y))
```
Use caution running this code as it will eventually use up all available memory on your machine.

The issue appears to be in PHASE 2 of the construction process.  The loop in question (starting at line 290) does not check the `failure` flag as one of the options for loop termination, it only checks `ishappy`.

**Comment by bhashemi** [comment]
I see the same problem in 'Chebfun2' with 'periodic' flag. Let's try a random 2D periodic bump function in the development branch:
```
gam = 10; x0 = 2*pi*rand-pi;  y0 = 2*pi*rand-pi;
 f = chebfun2(@(x,y) exp(-gam*((x-x0).^2+(y-y0).^2)),[-pi pi -pi pi],'periodic'); 
```
I see the same infinite loop as @gradywright has already noted:
```
Warning: Unresolved with maximum CHEBFUN length: 65537. 
> In chebfun2.constructor at 380
  In chebfun2.chebfun2>chebfun2.chebfun2 at 73 
Warning: Unresolved with maximum CHEBFUN length: 65537. 
> In chebfun2.constructor at 380
  In chebfun2.chebfun2>chebfun2.chebfun2 at 73 
Warning: Unresolved with maximum CHEBFUN length: 65537. 
> In chebfun2.constructor at 380
  In chebfun2.chebfun2>chebfun2.chebfun2 at 73 
Warning: Unresolved with maximum CHEBFUN length: 65537. 
> In chebfun2.constructor at 380
...
```

**Comment by ajt60gaibb** [comment]
It was a while ago, but I believe I fixed this in:  #1251.   #1251 is just 
waiting to be merged into development. I think I have already resolved 
all the issues that were brought up during code review, as I said it was 
a while ago.
